xapian-applications/omega/Makefile.am

```diff
@@ -164,12 +167,19 @@ omindex_SOURCES = omindex.cc myhtmlparse.cc htmlparse.cc index_file.cc mime.cc\
  runfilter.cc freemem.cc common/msvc_dirent.cc xpsxmlparse.cc common/str.cc\
  pkglibbindir.cc svgparse.cc tmpdir.cc urlencode.cc atomparse.cc xlsxparse.cc\
  opendocparse.cc common/keyword.cc msxmlparse.cc common/safe.cc timegm.cc\
- datetime.cc
+ datetime.cc common/closefrom.cc worker.cc worker_comms.cc
 if NEED_MKDTEMP
 omindex_SOURCES += portability/mkdtemp.cc
 endif
 omindex_LDADD = $(MAGIC_LIBS) $(XAPIAN_LIBS) $(ZLIB_LIBS) libutf8convert.la
 
+if OMINDEX_PDF
+pkglibbin_PROGRAMS += omindex_pdf
+noinst_HEADERS += handler_pdf.h
+omindex_pdf_SOURCES = assistant.cc worker_comms.cc handler_pdf.cc
+omindex_pdf_LDADD = -lpoppler-cpp $(XAPIAN_LIBS)
+endif
```

Comment:
This works for building, but the downside is that source files specific to this program will only be included in the tarball built by `make dist` if `OMINDEX_PDF` is true in the tree where `make dist` is run.  That's problematic as it makes it all too easy to miss source files from a release.

The better way to do this is to have all these lines unconditional except `pkglibbin_PROGRAMS += omindex_pdf`, and then list `omindex_pdf` in `EXTRA_PROGRAMS` - this tells automake that these programs may or may not be built, but it then knows they exist, and will include them during `make dist`.

Or scrap the `OMINDEX_PDF` conditional entirely and just `AC_SUBST` a variable which lists the modules to build, then you can substitute that variable in `pkglibbin_PROGRAMS` directly.

Reply:
I made some changes, but I am not sure if it is okay.